packages/melody-streams/src/component.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2018 trivago N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { enqueueComponent } from 'melody-idom';
+import { shallowEqual } from './util/shallowEqual';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+
+function Component(element) {
+    // part of the public API
+    this.el = element;
+    this.refs = {};
+    this.props = new BehaviorSubject({});
```

Comment:
Not sure, but `this.props` should be a plain object, otherwise this could break the dev tools.
Maybe we should name it `this.propsStream` and set `this.props` inside apply as soon as it receives new props

Reply:
Good point. Changed it.